### PR TITLE
Revamping the README

### PR DESCRIPTION
**Pull request by sfc-gh-ygu**
# Description

Let's make a great first impression! The README is the front page of our project and the first thing people see. We've simplified it to ensure that beginners can easily get started, and we've polished the formatting to make it look sharp and professional. This PR is all about making the README more user-friendly and visually appealing.

Screenshots after the change:
Screenshot 2024-08-04 at 11 08 54 PM

## Type of change
- [x] Doc change

# Checklist:
- [x] I have performed a self-review of my code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,184 +16,74 @@
...
+To connect from an engine like Spark, first create a catalog with these steps:
 ```bash
-$ export PRINCIPAL_TOKEN=ver:1-hint:1036-ETMsDgAAAY/GPANareallyverylongstringthatissecret
-
-$ curl -i -X POST -H "Authorization: Bearer $PRINCIPAL_TOKEN" -H 'Accept: application/json' -H 'Content-Type: application/json' \
-  http://${POLARIS_HOST:-localhost}:8181/api/management/v1/catalogs \
+# Generate a token for the root principal, replacing <CLIENT_ID> and <CLIENT_SECRET> with the values from the Polaris server output.
+export PRINCIPAL_TOKEN=$(curl -X POST http://localhost:8181/api/catalog/v1/oauth/tokens \
+  -d 'grant_type=client_credentials&client_id=<CLIENT_ID>&client_secret=<CLIENT_SECRET>&scope=PRINCIPAL_ROLE:ALL' \
+   | jq -r '.access_token')
```
Comment (sfc-gh-ygu) [inline README.md:77]: This script has been verified manually

File: README.md [hunk 2]
```diff
@@ -16,184 +16,74 @@
...
-```
-kubectl describe deployment polaris-deployment
-```
-
-## Creating a Catalog manually
-
-Before connecting with Spark, you'll need to create a catalog. To create a catalog, generate a token for the root
-principal:
-
+Building docs
```
Comment (sfc-gh-ygu) [inline README.md:63]: To simplify the README more, we can move the doc building part to a more detailed doc. WDYT?

File: README.md [hunk 3]
```diff
@@ -16,184 +16,74 @@
...
+- `docker run -p 8181:8181 localhost:5001/polaris:latest` - To run the image in standalone mode.
+- `docker compose up --build --exit-code-from regtest` - To run regression tests in a Docker environment.
+
+Running in Kubernetes
+- `./setup.sh` - To run Polaris as a mini-deployment locally. This will create two pods that bind themselves to port `8181`.
+- `kubectl get pods` - To check the status of the pods.
+- `kubectl get deployment` - To check the status of the deployment.
+- `kubectl describe deployment polaris-deployment` - To troubleshoot if things aren't working as expected.
+
+Polaris is organized into the following modules:
```
Comment (dimas-b) [inline README.md:58]: nit: WDYT about moving this to the top of the `Building and Running` section?

File: README.md (hunk 2 above)
Comment (dimas-b) [reply README.md:63]: From my POV it is good to have all essential building tips in the top-level README. I suppose contributors will be changing docs often.

**Review by dimas-b — APPROVED** [review]

**Review by eric-maynard — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -16,184 +16,76 @@
...
-The response output will contain an access token:
-
-```json
-{
-  "access_token": "ver:1-hint:1036-ETMsDgAAAY/GPANareallyverylongstringthatissecret",
-  "token_type": "bearer",
-  "expires_in": 3600
-}
+docker run -p 8080:80 -v ${PWD}:/spec docker.io/redocly/cli join spec/docs.yaml spec/polaris-management-service.yml spec/rest-catalog-open-api.yaml -o spec/index.yaml --prefix-components-with-info-prop title
+docker run -p 8080:80 -v ${PWD}:/spec docker.io/redocly/cli build-docs spec/index.yaml --output=docs/index.html --config=spec/redocly.yaml
```
Comment (annafil) [inline README.md:69]: LGTM! This is a much tidier way of presenting this info -- agree with you that the earlier links are no longer relevant now that everything is on a single GitHub Page. 🚢

**Review by collado-mike — APPROVED** [review]

**Pull request merged**